test/suites/dev-tanssi-relay/collator-assignment/test-configurable-priority.ts

```diff
@@ -0,0 +1,376 @@
+        });
+
+        it({
+            id: "E01",
+            title: "Set of Parachains should be sort by tip and truncated according to max cores allocated if we have less cores",
+            test: async function () {
+                const keyring = new Keyring({ type: "sr25519" });
+                const alice = keyring.addFromUri("//Alice", { name: "Alice default" });
+
+                const collatorAssignmentBefore = (
+                    await polkadotJs.query.tanssiCollatorAssignment.collatorContainerChain()
+                ).toJSON();
+                expect(sortCollatorAssignment(collatorAssignmentBefore)).to.be.deep.equal([
+                    2000, 2001, 2004, 2002, 2003,
+                ]);
+
+                // Record previous config value to restore later
+                const previousConfig = (await polkadotJs.query.collatorConfiguration.activeConfig()).toJSON();
+
+                // Let's change percentage of parachain to 30%
+                const tx = await polkadotJs.tx.sudo
+                    .sudo(polkadotJs.tx.collatorConfiguration.setMaxParachainCoresPercentage(300000000))
+                    .signAsync(alice);
+                await context.createBlock([tx]);
+
+                // Wait for two sessions for the effect
+                await jumpSessions(context, 2);
+
+                // Check the active assignment
+                const collatorAssignmentAfter = (
+                    await polkadotJs.query.tanssiCollatorAssignment.collatorContainerChain()
+                ).toJSON();
+                expect(sortCollatorAssignment(collatorAssignmentAfter)).to.be.deep.equal([2001, 2002, 2003, 2004]);
+
+                // Let's change percentage of parachain to 0
+                const zeroParachaintx = await polkadotJs.tx.sudo
+                    .sudo(polkadotJs.tx.collatorConfiguration.setMaxParachainCoresPercentage(0))
+                    .signAsync(alice);
+                await context.createBlock([zeroParachaintx]);
+
+                // Wait for two sessions for the effect
+                await jumpSessions(context, 2);
+
+                // Check the active assignment, we will not have any para id
+                const collatorAssignmentAtZeroPercent = (
+                    await polkadotJs.query.tanssiCollatorAssignment.collatorContainerChain()
+                ).toJSON();
+                expect(sortCollatorAssignment(collatorAssignmentAtZeroPercent)).to.be.deep.equal([2002, 2003, 2004]);
+
+                // Restore previous config
+                const restoringTx = await polkadotJs.tx.sudo
+                    .sudo(
+                        polkadotJs.tx.collatorConfiguration.setMaxParachainCoresPercentage(
+                            previousConfig.maxParachainCoresPercentage
+                        )
+                    )
+                    .signAsync(alice);
+                await context.createBlock([restoringTx]);
+
+                // Wait for two sessions for the effect
```

Comment:
this is asking for different test files if you need to restore the previous state, can you move the second test to a different file?

Reply:
I rearranged the tests so that now we do not have to reset the state in between.